src/tequila/circuit/gates.py

```diff
@@ -504,6 +504,68 @@ def SWAP(first: int, second: int, control: typing.Union[int, list] = None, power
     else:
         return GeneralizedRotation(angle=power * np.pi, control=control, generator=generator,
                                    eigenvalues_magnitude=0.25)
+        
+        
+def iSWAP(first: int, second: int, control: typing.Union[int, list] = None, power: float = 1.0, *args,
+         **kwargs) -> QCircuit:
+    """
+    Notes
+    ----------
+    iSWAP gate
+    .. math::
+        iSWAP = e^{i\\frac{\\pi}{4} (X \\otimes X + Y \\otimes Y )}
+
+    Parameters
+    ----------
+    first: int
+        target qubit
+    second: int
+        target qubit
+    control
+        int or list of ints
+    power
+        numeric type (fixed exponent) or hashable type (parametrized exponent)
+
+    Returns
+    -------
+    QCircuit
+
+    """
+
+    generator = paulis.from_string(f"X({first})X({second}) + Y({first})Y({second})")
+    return GeneralizedRotation(angle=power*(-np.pi/2), control=control, generator=generator,
```

Comment:
Looks good!
One issue that will arise with gradient compilation:

The generator here has 3 distinct eigenvalues (+2,-2 and 0), so it is not directly shift-rule differentiable. One can however fix this by exploiting the fact that one of the eigenvalues is 0. 

One can fix that in the following:
- write the generator as G = Gp + Gm, where Gp = G + P0 and Gm = G - P0 and P0 is the projector on the eigenspace with eigenvalue zero (in this case P0 = |00><00| + |11><11|) 
-

initialize like this
```
P0 = tq.paulis.Projector("|00>") + tq.paulis.Projector("|11>")
P0 = P0.map_qubits({0:first, 1:second})
```

The details behind this procedure: https://arxiv.org/abs/2011.05938

fortunately, the qubit excitation class then contains all the necessary technology. We just need to make sure, that specific optimized compilation routines are not called.

```python
gate = QubitExcitationImpl(angle=angle, target=generator.qubits, generator=generator, p0=p0, assume_real=assume_real, control=control, compile_option="vanilla")

return QCircuit.wrap_gate(gate)
```

... this should work (let me know if not)